R/copy-model-from.R

```diff
@@ -212,10 +221,53 @@ copy_control_stream <- function(.parent_model_path, .new_model_path, .overwrite,
 #' @return absolute file path to save new model to (without file extension)
 #' @keywords internal
 build_new_model_path <- function(.parent_mod, .new_model) {
+  if (is.null(.new_model)){
+    .new_model <- get_next_integer(.parent_mod)
+  }
+
+  if (length(.new_model) != 1) {
+    stop(paste(
+      glue("copy_model_from(.new_model) must be either a string or numeric, but got length {length(.new_model)}:\n\n"),
+      paste(.new_model, collapse = ", ")
+    ))
+  }
+
   if (!fs::is_absolute_path(.new_model)) {
     .new_model <- as.character(.new_model)
     .new_model <- file.path(get_model_working_directory(.parent_mod), .new_model)
     .new_model <- fs::path_norm(.new_model)
   }
   return(.new_model)
 }
+
+#' Check directory for model files with integer names
+#' and return the next integer. Used by copy_model_from().
+#' @importFrom stringr str_extract_all str_pad
+#' @keywords internal
+get_next_integer <- function(.parent_mod){
+  .dir <- dirname(get_model_path(.parent_mod))
+  .ext <- tools::file_ext(get_model_path(.parent_mod))
+
+  # find largest model in same dir with largest integer name
```

Comment:
`s/largest model/largest/`

"largest model" tripped me up given the "largest integer name" that follows, and seems redundant at best.